extensions/amp-story-subscriptions/0.1/test/test-amp-story-subscriptions.js

```diff
@@ -215,5 +220,66 @@ describes.realWin(
         SubscriptionsState.GRANTED
       );
     });
+
+    describe('skip button behaviors when embedded in a viewer', () => {
+      beforeEach(async () => {
+        const viewer = Services.viewerForDoc(env.ampdoc);
+        env.sandbox.stub(viewer, 'isEmbedded').withArgs().returns(true);
+        env.sandbox.stub(Services, 'viewerForDoc').returns(viewer);
+
+        storySubscriptions = new AmpStorySubscriptions(subscriptionsEl);
+        await nextTick(); // wait to make sure AmpStorySubscriptions is built.
+
+        storeService.dispatch(
+          Action.TOGGLE_SUBSCRIPTIONS_DIALOG_UI_STATE,
+          true
+        );
+
+        await afterRenderPromise(win);
+      });
+
+      it('should show skip button after delay', async () => {
+        await setTimeout(() => {
+          const buttonEl = this.win.document.querySelector(
+            'amp-subscriptions-dialog .i-amphtml-story-subscriptions-dialog-banner-button'
+          );
+          expect(buttonEl).to.have.class(
+            'i-amphtml-story-subscriptions-dialog-banner-button-visible'
+          );
+        }, SKIP_BUTTON_DELAY_DURATION);
```

Comment:
These will delay each test by 2 seconds, slowing down the test suite. You can fake the time using some library, you should be able to follow these examples in the codebase:

https://github.com/ampproject/amphtml/blob/83e7c80d7406923369ada8c1a7a5dc660ebdcad6/extensions/amp-sidebar/0.1/test/test-amp-sidebar.js#L219-L221

https://github.com/ampproject/amphtml/blob/83e7c80d7406923369ada8c1a7a5dc660ebdcad6/extensions/amp-sidebar/0.1/test/test-amp-sidebar.js#L251

But it looks like you might have to use the timer service over a setTimeout to get it to work:

https://github.com/ampproject/amphtml/blob/83e7c80d7406923369ada8c1a7a5dc660ebdcad6/extensions/amp-sidebar/0.1/amp-sidebar.js#L407